File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/shell/events/ShellEvent.java
package com.esofthead.mycollab.mobile.shell.events;

import com.esofthead.mycollab.eventmanager.ApplicationEvent;

/**
 * @author MyCollab Ltd.
 * @since 3.0
 */
public class ShellEvent {

    public static class GotoMainPage extends ApplicationEvent {

        public GotoMainPage(Object source, Object data) {
            super(source, data);
        }
    }

    public static class GotoCrmModule extends ApplicationEvent {

        public GotoCrmModule(Object source, Object data) {
            super(source, data);
        }
    }
}

File: mycollab-mobile/src/main/scala/com/esofthead/mycollab/mobile/shell/events/ShellEvent.scala
package com.esofthead.mycollab.mobile.shell.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

/**
 * @author myCollab Ltd
 * @since 5.2.1
 */
object ShellEvent {

    @SerialVersionUID(-551175801973985055L)
    class GotoLoginView(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoMainPage(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoCrmModule(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoProjectModule(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class GotoUserAccountModule(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class PushView(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

    class NavigateBack(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
}
